controllers/schedule_controller.go

```diff
@@ -0,0 +1,247 @@
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// ResourceType is the type to contain resource type string value
+type ResourceType string
+
+const (
+	// ManagedClusters resource type
+	ManagedClusters ResourceType = "managedClusters"
+	// Credentials resource type
+	Credentials ResourceType = "credentials"
+	// Resources related to applications and policies
+	Resources ResourceType = "resources"
+)
+
+var (
+	scheduleOwnerKey = ".metadata.controller"
+	apiGVString      = v1beta1.GroupVersion.String()
+	// mapping ResourceTypes to Velero schedule names
+	resourceTypes = map[ResourceType]string{
+		ManagedClusters: "acm-managed-clusters-schedule",
+		Credentials:     "acm-credentials-schedule",
+		Resources:       "acm-resources-schedule",
+	}
+)
+
+// BackupScheduleReconciler reconciles a BackupSchedule object
+type BackupScheduleReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=cluster.open-cluster-management.io,resources=backupschedules,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=cluster.open-cluster-management.io,resources=backupschedules/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=cluster.open-cluster-management.io,resources=backupschedules/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the BackupSchedule object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
+func (r *BackupScheduleReconciler) Reconcile(
+	ctx context.Context,
+	req ctrl.Request,
+) (ctrl.Result, error) {
+	scheduleLogger := log.FromContext(ctx)
+	backupSchedule := &v1beta1.BackupSchedule{}
+
+	if err := r.Get(ctx, req.NamespacedName, backupSchedule); err != nil {
+		// we can get not-found errors on deleted requests
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// retrieve the velero schedules (if any)
+	veleroScheduleList := veleroapi.ScheduleList{}
+	if err := r.List(ctx, &veleroScheduleList, client.InNamespace(req.Namespace), client.MatchingFields{scheduleOwnerKey: req.Name}); err != nil {
+		scheduleLogger.Error(
+			err,
+			"unable to list velero schedules for schedule",
+			"namespace", req.Namespace,
+			"name", req.Name,
+		)
+		return ctrl.Result{}, err
+	}
+
+	if len(veleroScheduleList.Items) == 0 {
+		if err := r.initVeleroSchedules(ctx, backupSchedule, r.Client); err != nil {
+			return ctrl.Result{}, err
+		}
+	} else {
```

Comment:
why the `else`? `golang` doesn't like `if-then-else` and in this case it's useless :)